### PR TITLE
v1.2 Add Patient & Edit Patient Command 

### PR DESCRIPTION
**Pull request by choowengyan**
This commit allows to add new patients to the PatientSync.

The parameters for `AddCommand` are: 
`patientHospitalId` , `name`, `preferredName`, `foodPreference`, `familyCondition`, `hobby`, `tags`

This commit also allows to edit an existing patient in the PatientSync using the `edit` command. 

To be done:
- [X]  Comand formatting
- [x]  Code testing (still on-going, need to fix some of the test cases)  
- [x]  UI Testing

Fixes issues: 
#4, #12

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -23,8 +23,7 @@ public class DeleteCommand extends Command {
             + "Parameters: INDEX (must be a positive integer)\n"
             + "Example: " + COMMAND_WORD + " 1";
 
-    public static final String MESSAGE_DELETE_PATIENT_SUCCESS = "Patient %1$s with ID %2$s has been "
-            + "successfully deleted.";
+    public static final String MESSAGE_DELETE_PATIENT_SUCCESS = "Deleted Patient: %1$s";
```
Comment (guohuang88) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java]: why is this changed back to the original success message?

**Review by guohuang88 — COMMENTED** [review]
Take a look at the comment I made!
May require one more reviewer to look through again as there are a lot of changes!

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (choowengyan) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java]: Might have accidentally accepted the past commit while resolving conflict. Have just updated it with the latest commit!

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 2]
```diff
@@ -6,10 +6,12 @@
 public class CliSyntax {
 
     /* Prefix definitions */
+    public static final Prefix PREFIX_PID = new Prefix("id/");
     public static final Prefix PREFIX_NAME = new Prefix("n/");
-    public static final Prefix PREFIX_PHONE = new Prefix("p/");
-    public static final Prefix PREFIX_EMAIL = new Prefix("e/");
-    public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
+    public static final Prefix PREFIX_PRENAME = new Prefix("p/");
```
Comment (SherisseTJW) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java]:
```suggestion
    public static final Prefix PREFIX_PREFERRED_NAME = new Prefix("p/");
```

Maybe spell it out fully so that its easier to know and remember what this Prefix is for

File: src/main/java/seedu/address/model/patient/FoodPreference.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    public FoodPreference(String food) {
+        requireNonNull(food);
+        checkArgument(isValidFood(food), MESSAGE_CONSTRAINTS);
+        foodPreference = food;
+    }
+
+    /**
+     * Returns true if a given string is a valid food details.
+     */
+    public static boolean isValidFood(String test) {
```
Comment (SherisseTJW) [inline src/main/java/seedu/address/model/patient/FoodPreference.java]:
```suggestion
    public static boolean isValidFoodPreference(String test) {
```

Maybe a better name so that the code is more readable

File: src/main/java/seedu/address/model/patient/Patient.java [hunk 4]
```diff
@@ -17,63 +17,82 @@
...
     private final Name name;
-    private final Phone phone;
-    private final Email email;
+    private final PreferredName preferredName;
 
     // Data fields
-    private final Address address;
+    private final FoodPreference foodPreference;
+    private final FamilyCondition familyCondition;
+    private final Hobby hobby;
```
Comment (SherisseTJW) [inline src/main/java/seedu/address/model/patient/Patient.java:27]: I think it may make more sense for these fields to be a `Set` as well... Since a person can have multiply food preferences, hobbies and family conditions.

**Review by SherisseTJW — CHANGES_REQUESTED** [review]
Generally ok, some suggestions for better names.

The main thing to discuss would be whether to make Food Preferences, Hobbies and Family Condition Sets, so that the user can input multiple

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 5]
```diff
@@ -6,10 +6,12 @@
...
+    public static final Prefix PREFIX_FOOD = new Prefix("f/");
```
Comment (ZhiWei1010) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java]: Similar to @SherisseTJW 's comment, maybe can spell out to be PREFIX_FOOD_PREFERENCE instead to deter confusion with other food related information such as FOOD_ALLERGIES.

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 6]
```diff
@@ -6,10 +6,12 @@
...
+    public static final Prefix PREFIX_FAMILY = new Prefix("c/");
```
Comment (ZhiWei1010) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java]: Similarly, suggest to rename PREFIX_FAMILY to. PREFIX_FAMILY_CONDITION to deter confusion with other family related information and better scalability.

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 5 above)
Comment (choowengyan) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java]: Have been updated to `PREFIX_FOOD_PREFERENCE`

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 6 above)
Comment (choowengyan) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java]: Have been updated to `PREFIX_FAMILY_CONDITION`

File: src/main/java/seedu/address/model/patient/FoodPreference.java (hunk 3 above)
Comment (choowengyan) [reply src/main/java/seedu/address/model/patient/FoodPreference.java]: Have been updated! thanks for the suggestion!

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 2 above)
Comment (choowengyan) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java]: Have been updated to `PREFIX_PREFERRED_NAME`

File: src/main/java/seedu/address/model/patient/Patient.java (hunk 4 above)
Comment (SherisseTJW) [reply src/main/java/seedu/address/model/patient/Patient.java:27]: After offline discussion, a separate Issue and PR will be created to handle this enhancement.

**Review by SherisseTJW — APPROVED** [review]
LGTM
